deploy/eck-integrations/templates/eck-agent/elastic-agent-daemonset.yaml

```diff
@@ -0,0 +1,95 @@
+---
+apiVersion: agent.k8s.elastic.co/v1alpha1
+kind: Agent
+metadata:
+  name: eck-agent-daemonset
+  namespace: {{ .namespace | default $.Release.Namespace | quote }}
+  labels:
+    {{- include "elasticagent.labels" $ | nindent 4 }}
+  annotations:
+    {{- with .Values.eck_agent.annotations }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+spec:
+  version: {{ required "An Elastic Agent version is required" .Values.eck_agent.version }}
+  configRef:
+    secretName: eck-agent-daemonset
+  daemonSet:
+    podTemplate:
+      metadata:
+        annotations:
+          checksum/config: {{ include (print $.Template.BasePath "/eck-agent/elastic-agent-daemonset-secret.yaml") . | sha256sum }}
+      spec:
+        automountServiceAccountToken: true
+        serviceAccountName: eck-agent
+        containers:
+        - name: agent
+          securityContext:
+            runAsUser: 0
+            {{- if eq .Values.cloudDefend.enabled true}}
+            capabilities:
+              add:
+                - BPF # (since Linux 5.8) allows loading of BPF programs, create most map types, load BTF, iterate programs and maps.
+                - PERFMON # (since Linux 5.8) allows attaching of BPF programs used for performance metrics and observability operations.
+                - SYS_RESOURCE # Allow use of special resources or raising of resource limits. Used by 'Defend for Containers' to modify 'rlimit_memlock'
+            {{- end}}
+          resources:
+            limits:
+              memory: 700Mi
+            requests:
+              cpu: 100m
+              memory: 400Mi
```

Comment:
I think also the limits should be configurable with default values

Reply:
this is now configurable in values.yaml under eck_agent.presets.presetName, and the user can change it at will